Sources/ProjectDescription/FileCodeGen.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  FileCodeGen.swift
+//  FileCodeGen
+//
+//  Created by Mahadevaiah, Pavan | Pavan | ECMPD on 2021/09/14.
+//
+
+import Foundation
+
+/// FileCodeGen: Soure file code generation attribues
+///
+/// - `public`:  public codegen attribute  `settings = {ATTRIBUTES = (codegen, )`}
+/// - `private`:  private codegen attribute  `settings = {ATTRIBUTES = (private_codegen, )}`
+/// - `project`:  project codegen attribute  `settings = {ATTRIBUTES = (project_codegen, )}`
+/// - `disabled`:  disabled codegen attribute  `settings = {ATTRIBUTES = (no_codegen, )}`
+///
+public enum FileCodeGen: Codable, Equatable {
```

Comment:
To support Xcode 12.5.1 (which doesn't yet support enum codable synthesis), we'll need to make this enum have a string raw value.

```swift
public enum FileCodeGen: String, Codable, Equatable
```